### ISSUE TITLE
Track external (pkgconfig) dependencies for custom setup script

### ISSUE THREAD

**Issue by k0ral** [haskell/cabal#3654]
The [`haskell-gi`-based packages][1] are heavily relying on a custom `Setup.hs` that generates the haskell bindings to `glib`-based C libraries. Consequently, this process depends on the C library at setup-time, but it seems cabal only makes it possible to express a build-time dependency (through `pkgconfig-depends` within the `library` section). In systems like NixOS, this distinction matters a lot.
Are you planning on supporting this use case in the future ? Otherwise, how would you suggest to overcome this untracked dependency issue ?

Thank you.

[1]: https://github.com/haskell-gi/haskell-gi

**Comment by 23Skidoo** [comment]
Have you tried using `build-type: Configure` and [generating a .buildinfo file](https://downloads.haskell.org/~ghc/7.0.4/docs/html/Cabal/authors.html#system-dependent)?

**Comment by ezyang** [comment]
I don't think that helps; Configure runs a configure script which is still just a build-time dependency checker. So the answer is, no, we're pretty bad at tracking runtime deps.
